src/fides/api/custom_types.py

```diff
@@ -0,0 +1,56 @@
+"""Logic related to sanitizing and validating user application input."""
+from html import escape
+from re import compile as regex
+from typing import Generator
+
+
+class SafeStr(str):
+    """
+    This class is designed to be used in place of the `str` type
+    any place where user input is expected.
+
+    The validation applied here does the typical sanitization/cleanup
+    that is required when dealing with user-supplied input.
+    """
+
+    @classmethod
+    def __get_validators__(cls) -> Generator:  # pragma: no cover
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value: str) -> str:
+
+        # HTML Escapes
+        value = escape(value)
+
+        if len(value) > 500:
+            raise ValueError("Value must be 500 characters or less.")
```

Comment:
Some connectors require inputs longer than 500 characters as part of the setup, what is the reasoning behind limiting the input length?

Reply:
User input sanitization best practices. We can avoid using this for places that we generally trust I believe, like configuring connectors